content/articles/2018-02-usethis-1-3-0.Rmd

```diff
@@ -0,0 +1,106 @@
+#> ✔ Writing to user git config file
+#> $user.name
+#> [1] "Jane Doe"
+#> 
+#> $user.email
+#> [1] "jane@example.com"
+```
+
+[`browse_github_pat()`](http://usethis.r-lib.org/reference/browse_github_pat.html) takes you to a pre-filled web form to generate a GitHub personal access token (PAT). You are instructed how to store this as an environment variable, which usethis can then use downstream to, e.g., create new GitHub repositories on your behalf.
+
+[`create_from_github()`](http://usethis.r-lib.org/reference/create_from_github.html) is not new, but is newly capable of both the ["clone"](https://help.github.com/articles/cloning-a-repository/) or ["fork and clone"](https://help.github.com/articles/fork-a-repo/) workflows, with either the https or ssh protocols. It gains optional arguments for direct provision of the PAT, Git credentials, and GitHub host. usethis can now facilitate all the major workflows:
+
+  * GitHub first, local project/repo second: [`create_from_github()`](http://usethis.r-lib.org/reference/create_from_github.html) does this, with and without forking. In the case of a fork, both the `origin` and `upstream` remotes are configured, so you are ready to stay in sync with the upstream repo.
+  * Local project/repo first, GitHub last: [`use_github()`](http://usethis.r-lib.org/reference/use_github.html) does this.
+
+## Distribution of course materials
+
+Many courses and workshops kick-off with the participants downloading a set of files from the instructor, such as slides or code examples. We have found this to be a reliable source of aggravation, due to the diversity of the learners' file system knowledge and habits. You cannot assume that everyone will quickly download an entire set of files, and only those files, to a well-named folder, that they can find in their file browser. [`use_course()`](http://usethis.r-lib.org/reference/use_course.html) is a targeted solution to this problem, when time is of the essence.
+
+[`use_course()`](http://usethis.r-lib.org/reference/use_course.html) accepts a link and interactively obtains user's permission to download a ZIP archive and unpack it to a deliberately named folder on user's desktop. If an RStudio Project file is found, the new Project is launched and, otherwise, the folder is opened in the file manager, e.g. Finder or File Explorer.
+
+Here's an example where the bit.ly shortlink points to the usethis source on GitHub.
+
+``` r
+usethis::use_course("bit.ly/usethis-shortlink-example")
+#> ✔ Downloading ZIP file to '/Users/jenny/Desktop/r-lib-usethis-v1.2.0-93-g7d1c255.zip'
+#> ✔ Unpacking ZIP file into '/Users/jenny/Desktop/r-lib-usethis-7d1c255/' (272 files extracted)
+```
+
+## Community-specific conventions
+
+The development conventions for [tidyverse packages](https://github.com/tidyverse/) are gradually being codified into a set of [`use_tidy_*()`](http://usethis.r-lib.org/reference/tidyverse.html) helper functions.
```

Comment:
Do we want to be a bit more explicit about what is/is not appropriate for external use? For example, I think the contributing guidelines use tidyverse-specific language, which would be thoroughly confusing should anyone else unwittingly keep that language in their own CONTRIBUTING.md.

Reply:
Current version clarifies this.